src/push/Permission.js

```diff
@@ -62,13 +66,14 @@ export default class Permission {
             this._win.Notification &&
             this._win.Notification.requestPermission
         ) {
-            /* Chrome 23+ */
-            this._win.Notification
-                .requestPermission()
-                .then(resolve)
-                .catch(function() {
+            /* Safari 12+ */
+            request = this._win.Notification.requestPermission(resolve);
+            if (request && request.then) {
```

Comment:
It possibly makes it redundant, but I just prefer not to assume (as Safari is already not conforming to one standard) that when Safari _does_ return something, it will be a Promise.